### PR TITLE
Feature/js api dev tools recording compatibility

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/common/assets/js/api/core/commands.js [hunk 1]
```diff
@@ -282,6 +282,8 @@ export default class Commands extends elementorModules.Module {
 
 		this.afterRun( command );
 
+		this.trigger( 'run:after', component, command, args );
```
Comment (matipojo) [inline core/common/assets/js/api/core/commands.js]: Change also 'run' to 'run:before' + BC

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/module/base.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
+class ModuleBase {
```
Comment (matipojo) [inline core/common/assets/js/api/core/module/base.js]: Can you try replace the original module wiht the ES6 version?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/backwards-compatibility/commands.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export default class CommandsBackwardsCompatibility extends elementorModules.Module {
+	__construct() {
+		this.onOrig = this.on;
+	}
+
+	on = ( eventName, callback ) => {
+		if ( 'run' === eventName ) {
+			let componentName = this.getConstructorID();
+
+			componentName = componentName.replace( /^./, ( val ) => val.toLowerCase() );
```
Comment (matipojo) [inline core/common/assets/js/api/core/backwards-compatibility/commands.js]: Explain

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

File: core/common/assets/js/api/core/backwards-compatibility/commands.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+export default class CommandsBackwardsCompatibility extends elementorModules.Module {
+	__construct() {
+		this.onOrig = this.on;
+	}
+
+	on = ( eventName, callback ) => {
```
Comment (kobizz) [inline core/common/assets/js/api/core/backwards-compatibility/commands.js]:
```suggestion
	on( eventName, callback ) {
```
Why it's in a different syntax?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/backwards-compatibility/commands.js (hunk 4 above)
Comment (matipojo) [reply core/common/assets/js/api/core/backwards-compatibility/commands.js]: Because it extends a non ES6 object, it doesn't work.

Comment (kobizz) [reply core/common/assets/js/api/core/backwards-compatibility/commands.js]: It doesn't make sense. A lot of our modules extend exactly this none-ES6 module, and they work correctly.

Comment (iNewLegend) [reply core/common/assets/js/api/core/backwards-compatibility/commands.js]: @kobizz, you have better solution?

Comment (matipojo) [reply core/common/assets/js/api/core/backwards-compatibility/commands.js:6]: @kobizz, we already tried, it doesn't work.
Do you want to check it yourself?

**Review by kobizz — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Comment by iNewLegend** [comment]
@matipojo, please re-approve.

**Review by KingYes — APPROVED** [review]

**Pull request merged**